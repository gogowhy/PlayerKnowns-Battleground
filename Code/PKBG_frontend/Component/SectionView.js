import React, {Component} from 'react';
import { 
    StyleSheet, 
    Text, 
    View,
    SectionList,
    Dimensions}
from 'react-native';

var {width} = Dimensions.get('window');

export default class SectionView extends Component {
    constructor(props) { //构造器
        super(props);
        this.state = {
            refreshing: false,  //是否刷新,通过更改此属性来控制是否刷新
            sections:[   //数据源
                {team:'A', data:[
                    {name : 'xie yihan' , isReady : true},
                    {name : 'wang haoyu' , isReady : false}]},
                {team:'B', data:[
                    {name : 'qi peng' , isReady : true},
                    {name : 'zhou yifan' , isReady : true}]}
            ]
        }
    }

    render() {

        return (
          <View style={{flex: 1}}>
              <SectionList
                    style={{marginTop:10,width:width-30}}
                    renderSectionHeader={this._sectionComp} //区头
                    renderItem={this._renderItem}   //cell
                    sections={this.state.sections}     //数据源
                    ItemSeparatorComponent={() => <View style={{backgroundColor:'white',height:1}}></View>}  //分割线
                    stickySectionHeadersEnabled={false}  //设置区头是否悬浮在屏幕顶部,默认是true
                    initialNumToRender = {2} //指定一开始渲染的元素数量，最好刚刚够填满一个屏幕，这样保证了用最短的时间给用户呈现可见的内容
                    keyExtractor = {this._extraUniqueKey}
                    setVerticalScrollBarEnabled = {false}
                    setFastScrollEnabled = {false}
              />
          </View>
        );
    }

    _renderItem = (info) => {
        var txt1 = info.item.name;
        if(info.item.isReady == true)
        {
            var txt2 = 'Ready';
        }
        else{
            txt2 = 'UnReady';
        }
        return (
            <View style={styles.playerCell}>
                <View style={styles.name}>
                    <Text style={styles.cellText}>{txt1}</Text>
                </View>
                <View style={styles.isReady}>
                    <Text style={styles.cellText}>{txt2}</Text>
                </View>
            </View>
            )
    }

    _sectionComp = (info) => {
        var txt = ' Team '+info.section.team;
        return (
            <View style={{flex: 1, height: 25, backgroundColor: '#11ffff', justifyContent: 'center'}}>
                <Text style={styles.SectionHeader}>{txt}</Text>
            </View>
        )
    }

    _extraUniqueKey =(item)=> {
        return "index"+item;
    };
}

const styles = StyleSheet.create({
    playerCell: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#607B8B', 
    },
    name: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
    isReady: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
    cellText: {
        height: 30,
        textAlignVertical: 'center', 
        color: '#fff', 
        fontSize: 15,
        marginLeft: 10,
        marginRight: 10,
    },
    SectionHeader: { 
        height: 30,
        textAlignVertical: 'center',
        backgroundColor: '#473C8B',
        color: 'white',
        fontSize: 20
    },
})
    
