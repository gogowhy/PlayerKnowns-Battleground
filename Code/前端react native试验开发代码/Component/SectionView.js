import React, {Component} from 'react';
import { 
    StyleSheet, 
    Text, 
    View,
    SectionList,
    Dimensions}
from 'react-native';

var {width} = Dimensions.get('window');

export default class SectionView extends Component {
    constructor(props) { //构造器
        super(props);
        this.state = {
            refreshing: false,  //是否刷新,通过更改此属性来控制是否刷新
            sections:[   //数据源
                {team:'Team A', data:[
                    {name : 'xie yihan' , isReady : true},
                    {name : 'wang haoyu' , isReady : false}]},
                {team:'Team B', data:[
                    {name : 'qi peng' , isReady : true},
                    {name : 'zhou yifan' , isReady : true}]}
            ]
        }
    }

    render() {

        return (
          <View style={{flex: 1}}>
              <SectionList
                    style={{marginTop:10,width:width-30}}
                    renderSectionHeader={this._sectionComp} //区头
                    renderItem={this._renderItem}   //cell
                    sections={this.state.sections}     //数据源
                    ItemSeparatorComponent={() => <View style={{backgroundColor:'white',height:1}}></View>}  //分割线
                    stickySectionHeadersEnabled={false}  //设置区头是否悬浮在屏幕顶部,默认是true
                    initialNumToRender = {2} //指定一开始渲染的元素数量，最好刚刚够填满一个屏幕，这样保证了用最短的时间给用户呈现可见的内容
                    setVerticalScrollBarEnabled = {false}
                    setFastScrollEnabled = {false}
              />
          </View>
        );
    }

    _renderItem = (info) => {
        var txt1 = info.item.name;
        if(info.item.isReady == true)
        {
            var txt2 = 'Ready';
        }
        else{
            txt2 = 'UnReady';
        }
        return (
            <View style={styles.playerCell}>
                <Text style={styles.cellText}>{txt1}</Text>
                <Text style={styles.cellText}>{txt2}</Text>
            </View>
            )
    }

    _sectionComp = (info) => {
        var txt = ' '+info.section.team;
        return <Text style={styles.SectionHeader}>{txt}</Text>
    }
}

const styles = StyleSheet.create({
    playerCell: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around',
        alignItems: 'center',
        backgroundColor: '#607B8B', 
    },
    cellText: {
        height: 30,
        textAlignVertical: 'center', 
        color: '#fff', 
        fontSize: 15,
        marginLeft: 10,
        marginRight: 10,
    },
    SectionHeader: { 
        height: 30,
        textAlignVertical: 'center',
        backgroundColor: '#473C8B',
        color: 'white',
        fontSize: 20
    },
})
    
