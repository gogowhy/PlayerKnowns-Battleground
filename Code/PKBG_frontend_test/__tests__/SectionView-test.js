/**
 * @format
 */

import 'react-native';
import React from 'react';
import SectionView from '../Component/SectionView';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<SectionView />).toJSON();
  expect(tree).toMatchSnapshot();
});
/*
import React from 'react';
import Intro from '../Intro';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});*/
