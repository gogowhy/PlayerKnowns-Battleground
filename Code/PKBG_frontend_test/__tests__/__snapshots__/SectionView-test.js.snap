// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "isReady": true,
              "name": "xie yihan",
            },
            Object {
              "isReady": false,
              "name": "wang haoyu",
            },
          ],
          "team": "A",
        },
        Object {
          "data": Array [
            Object {
              "isReady": true,
              "name": "qi peng",
            },
            Object {
              "isReady": true,
              "name": "zhou yifan",
            },
          ],
          "team": "B",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={2}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "isReady": true,
              "name": "xie yihan",
            },
            Object {
              "isReady": false,
              "name": "wang haoyu",
            },
          ],
          "team": "A",
        },
        Object {
          "data": Array [
            Object {
              "isReady": true,
              "name": "qi peng",
            },
            Object {
              "isReady": true,
              "name": "zhou yifan",
            },
          ],
          "team": "B",
        },
      ]
    }
    setFastScrollEnabled={false}
    setVerticalScrollBarEnabled={false}
    stickyHeaderIndices={Array []}
    stickySectionHeadersEnabled={false}
    style={
      Object {
        "marginTop": 10,
        "width": 720,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#11ffff",
              "flex": 1,
              "height": 25,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "#473C8B",
                "color": "white",
                "fontSize": 20,
                "height": 30,
                "textAlignVertical": "center",
              }
            }
          >
             Team A
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#607B8B",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "height": 30,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "textAlignVertical": "center",
                  }
                }
              >
                xie yihan
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 15,
                    "height": 30,
                    "marginLeft": 10,
                    "marginRight": 10,
                    "textAlignVertical": "center",
                  }
                }
              >
                Ready
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "height": 1,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
